lib/features/email/domain/usecases/view_attachment_for_web_interactor.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:async';
+
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:model/download/download_task_id.dart';
+import 'package:model/email/attachment.dart';
+import 'package:tmail_ui_user/features/email/domain/state/download_attachment_for_web_state.dart';
+import 'package:tmail_ui_user/features/email/domain/state/view_attachment_for_web_state.dart';
+import 'package:tmail_ui_user/features/email/domain/usecases/download_attachment_for_web_interactor.dart';
+
+class ViewAttachmentForWebInteractor {
+  ViewAttachmentForWebInteractor(this._downloadAttachmentForWebInteractor);
+
+  final DownloadAttachmentForWebInteractor _downloadAttachmentForWebInteractor;
+
+  Stream<Either<Failure, Success>> execute(
+    DownloadTaskId taskId,
+    Attachment attachment,
+    AccountId accountId,
+    String baseDownloadUrl,
+    StreamController<Either<Failure, Success>> onReceiveController,
+  ) =>
+      _downloadAttachmentForWebInteractor
+          .execute(
+            taskId,
+            attachment,
+            accountId,
+            baseDownloadUrl,
+            onReceiveController)
+          .map(
+            (result) => result.fold(
+              (failure) {
+                if (failure is DownloadAttachmentForWebFailure) {
+                  return left(ViewAttachmentForWebFailure(
+                    attachment: attachment,
+                    taskId: failure.taskId,
+                    exception: failure.exception,
+                  ));
+                }
+
+                return left(failure);
+              },
+              (success) {
+                if (success is StartDownloadAttachmentForWeb) {
+                  return right(StartViewAttachmentForWeb(
+                    success.taskId,
+                    success.attachment,
+                  ));
+                }
+
+                if (success is DownloadingAttachmentForWeb) {
+                  return right(ViewingAttachmentForWeb(
+                    success.taskId,
+                    success.attachment,
+                    success.progress,
+                    success.downloaded,
+                    success.total,
+                  ));
+                }
+
+                if (success is DownloadAttachmentForWebSuccess) {
+                  return right(ViewAttachmentForWebSuccess(
+                    success.taskId,
+                    success.attachment,
+                    success.bytes,
+                  ));
+                }
+
+                return right(success);
```

Comment:
Please use the keyword `Right/Left` is printed in capital letters for synchronization throughout the app, avoiding confusion for new users.

Reply:
I pushed this fixup at PR #2642 , please review there as this file wasn't created in this PR